homeassistant/components/webhook/__init__.py

```diff
@@ -111,7 +127,21 @@ async def async_handle_webhook(
         _LOGGER.debug("%s", content)
         return Response(status=HTTPStatus.OK)
 
-    if webhook["local_only"]:
+    if method_name not in webhook["allowed_methods"]:
+        _LOGGER.warning(
+            "Webhook %s only supports %s methods but %s was received from %s",
+            webhook_id,
+            ",".join(webhook["allowed_methods"]),
+            method_name,
+            received_from,
+        )
+        if method_name == METH_HEAD:
+            # Allow websites to verify that the URL exists.
+            # See https://github.com/home-assistant/core/pull/26299
```

Comment:
The comments by themselves should carry the reasoning. In this case, I think it already does.

```suggestion
```